api/query/search.go

```diff
@@ -85,82 +85,112 @@ func (sh structuredSearchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	// Check if the query is a simple (empty/just True, or test name only) query
+	// Prepare logging.
+	ctx := sh.api.Context()
+	logger := shared.GetLogger(ctx)
+
 	var simpleQ TestNamePattern
+
+	r2 := r.Clone(r.Context())
+	r2url := *r.URL
+	r2.URL = &r2url
+	r2.Method = "GET"
+	q := r.URL.Query()
+	q.Add("q", simpleQ.Pattern)
```

Comment:
nit: for the various query string parameter keys, we should declare constants

Reply:
Yes, that would be useful, and might be best stored somewhere like `shared/params.go`. Maybe this is a possible code cleanup issue? I'm not sure the scope of this change, but it would probably be worth handling in all places.